morebits.js

```diff
@@ -2625,6 +2601,45 @@ Morebits.wiki.page = function(pageName, currentAction) {
 		}
 	};
 
+	/**
+	 * Marks the page as patrolled, using rcid (if available) or revid
+	 *
+	 * Patrolling as such doesn't need to rely on loading the page in
+	 * question; simply passing a revid to the API is sufficient, so in
+	 * those cases just using Morebits.wiki.api is probably preferable.
+	 *
+	 * No error handling since we don't actually care about the errors
+	 */
+	this.patrol = function() {
+		if (!Morebits.userIsSysop && !Morebits.userIsInGroup('patroller')) {
```

Comment:
Most wikis don't have the patroller group. Would it be worth adding a `mw.config.get('wgDbName') === 'enwiki'` check here?

Reply:
It's otherwise limited to `sysop` though, no?  `Morebits.userIsInGroup('patroller')` will just return `false` if the group doesn't exist, so there wouldn't be any difference.  The issue would be if a wiki **did** have a patroller group but it didn't have `patrol`...

Reply:
Looking at the code, looks like that's the case for most wikis, though I'm surprised. At enwiki, it was available to all autoconfirmed users before creation of the patroller group, but it turns out [that itself was an exception](https://phabricator.wikimedia.org/T14007) made many years back. 

There are a few wikis though which give patrol to autoconfirmed users, where this code will fail, but I guess that can be handled later along with i18n.

Reply:
Oh yeah, the good ol' days!

The other option would be `mw.user.getRights().then(function (rights)...` but I really don't think that's an improvement.